simphony/io/file_lattice.py

```diff
@@ -0,0 +1,378 @@
+from simphony.core.cuba import CUBA
+from simphony.cuds.lattice import LatticeNode
+
+import tables
+
+import numpy
+import copy
+
+
+class FileLattice:
+    """FileLattice object to use in file lattices.
+
+    Parameters
+    ----------
+    file : CudsFile
+        reference to PyTables file where the lattice is or will be located
+    name : str
+        name of the lattice
+    type: string
+        Bravais lattice type (should agree with the base_vect below).
+    base_vect: D x float
+        defines a Bravais lattice (an alternative for primitive vectors).
+    size: D x size
+        number of lattice nodes (in the direction of each axis).
+    origin: D x float
+
+    Returns
+    ----------
+    FileLattice
+        The lattice newly added to the file or existing in the file.
+
+    """
+
+    class _ColDescr(tables.IsDescription):
+        pass
+
+    def __init__(self, file, name, type, base_vect, size, origin):
+        # Construct default FileLattice
+        self._file = file._file
+        self._group = file._file.root.lattice
+        self._name = name
+        self._type = type
+        self._base_vect = base_vect
+        self._size = size
+        self._origin = origin
+        self._maskname = name+'_bitmask'
+        self._mask_size = 8
+
+        # Construct default rowclass based on lattice metadata
+        self._construct_default_rowclass()
+
+        if self._name not in self._group:
+            # Create default table to hold lattice data
+            self._create_table()
+            self._table.attrs.type = self._type
+            self._table.attrs.base_vect = self._base_vect
+            self._table.attrs.size = self._size
+            self._table.attrs.origin = self._origin
+
+        else:
+            # Return reference to existing table
+            self._table = self._file.root.lattice._f_getChild(self._name)
+            self._type = self._table.attrs.type
+            self._base_vect = self._table.attrs.base_vect
+            self._size = self._table.attrs.size
+            self._origin = self._table.attrs.origin
+            # Update column description class.
+            data = {}
+            for key in self._table.coldescrs:
+                if not key == 'INDEX':
+                    data[eval('CUBA.'+key)] = self._table[:][key][0]
+            if not data == {}:
+                self._update_coldescr(data, data.keys())
+
+    def _construct_default_rowclass(self):
+        """ Construct _ColDescr class with one column, so that
+        PyTables table for the lattice can be generated. Empty table
+        cannot be created.
+        TODO: Find a more compact (and still meaningful) way of
+        generating a dense table.
+        """
+        self._ColDescr = type("IsDescription", (tables.IsDescription,),
+                              {'INDEX': tables.Int32Col(pos=0,
+                               shape=len(self._base_vect))})
+
+    def _update_coldescr(self, data, keywords):
```

Comment:
One can retrieve the full `DataContainer` table description from https://github.com/simphony/simphony-common/blob/master/simphony/io/data_container_description.py#L5